tick/inference/hawkes_sumgaussians.py

```diff
@@ -0,0 +1,404 @@
+import math
+import numpy as np
+from scipy.stats import norm
+from tick.inference.base import LearnerHawkesNoParam
+from tick.optim.solver.base.utils import relative_distance
+from .build.inference import HawkesSumGaussians as _HawkesSumGaussians
+
+
+class HawkesSumGaussians(LearnerHawkesNoParam):
+    """A class that implements parametric inference for Hawkes processes
+    with parametrisation of the kernels as sum of Gaussian basis functions
+    and a mix of Lasso and group-lasso regularization
+
+    Hawkes processes are point processes defined by the intensity:
+
+    .. math::
+        \\forall i \\in [1 \\dots D], \\quad
+        \\lambda_i(t) = \\mu_i + \\sum_{j=1}^D
+        \\sum_{t_k^j < t} \\phi_{ij}(t - t_k^j)
+
+    where
+
+    * :math:`D` is the number of nodes
+    * :math:`\mu_i` are the baseline intensities
+    * :math:`\phi_{ij}` are the kernels
+    * :math:`t_k^j` are the timestamps of all events of node :math:`j`
+
+    and with an parametrisation of the kernels as sum of Gaussian basis 
+    functions
+
+    .. math::
+        \phi_{ij}(t) = \sum_{m=1}^M \\alpha^{ij}_m f (t - t_m), \\quad
+        f(t) = (2 \\pi \\sigma^2)^{-1} \exp(- t^2 / (2 \\sigma^2))
+
+    In our implementation we denote:
+
+    * Integer :math:`D` by the attribute `n_nodes`
+    * Vector :math:`\mu \in \mathbb{R}^{D}` by the attribute
+      `baseline`
+    * Vector :math:`(t_m) \in \mathbb{R}^{M}` by the variable
+      `means_gaussians`
+    * Number :math:`\\sigma` by the variable `std_gaussian`
+    * Tensor 
+      :math:`A = (\\alpha^{ij}_m)_{ijm} \in \mathbb{R}^{D \\times D \\times M}`
+      by the attribute `amplitudes`
+    * Number :math:`\\beta \in \mathbb{R}` by the parameter `decay`. This
+      parameter is given to the model
```

Comment:
it would be cool if `means_gaussians` and `std_gaussian` was attributes of the model. It could be done with a property.